Permission denied through skaffold

Hi, I'am using the version 0.20.0.

I have a strange behavior with the `build` stage. Here is my Dockerfile : 

```
FROM node as build

COPY package.json package.json
RUN npm install
COPY . .
RUN npm run tsc

FROM node:alpine
COPY --from=build ./dist .
COPY package.json package.json
RUN npm install --prod
CMD ["node","index.js"]
```

When I build this Dockerfile using my local Docker, it works without any problem.
But, If I launch a `skaffold dev` the build fails with this error : 

```
Step 5/10 : RUN npm run tsc
 ---> Running in 0c947ae2b542

> servicets@1.0.0 tsc /
> ./node_modules/typescript/bin/tsc

sh: 1: ./node_modules/typescript/bin/tsc: Permission denied
npm ERR! code ELIFECYCLE
npm ERR! errno 126
npm ERR! servicets@1.0.0 tsc: `./node_modules/typescript/bin/tsc`
npm ERR! Exit status 126
npm ERR!
npm ERR! Failed at the servicets@1.0.0 tsc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-01-15T13_37_07_099Z-debug.log
```

I have a permission denied through Skaffold that I don't have through my local docker.

The only workarround I have found is to add 

     RUN chmod a+x ./node_modules/typescript/bin/tsc

Just before `RUN npm run tsc`

I don't understand why I have these different behaviors

Thank you

--- a/pkg/skaffold/util/tar.go
+++ b/pkg/skaffold/util/tar.go
@@ -22,6 +22,7 @@ import (
 	"io"
 	"os"
 	"path/filepath"
+	"runtime"
 
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
@@ -102,7 +103,7 @@ func addFileToTar(root string, src string, dst string, tw *tar.Writer) error {
 		}
 		tarHeader.Name = tarPath
 
-		if err := tw.WriteHeader(tarHeader); err != nil {
+		if err := writeHeader(tw, tarHeader); err != nil {
 			return err
 		}
 	case mode.IsRegular():
@@ -112,7 +113,7 @@ func addFileToTar(root string, src string, dst string, tw *tar.Writer) error {
 		}
 		tarHeader.Name = tarPath
 
-		if err := tw.WriteHeader(tarHeader); err != nil {
+		if err := writeHeader(tw, tarHeader); err != nil {
 			return err
 		}
 
@@ -140,7 +141,7 @@ func addFileToTar(root string, src string, dst string, tw *tar.Writer) error {
 			return err
 		}
 		tarHeader.Name = tarPath
-		if err := tw.WriteHeader(tarHeader); err != nil {
+		if err := writeHeader(tw, tarHeader); err != nil {
 			return err
 		}
 	default:
@@ -150,9 +151,30 @@ func addFileToTar(root string, src string, dst string, tw *tar.Writer) error {
 		if err != nil {
 			return err
 		}
-		if err := tw.WriteHeader(tarHeader); err != nil {
+		if err := writeHeader(tw, tarHeader); err != nil {
 			return err
 		}
 	}
 	return nil
+}
+
+// Code copied from https://github.com/moby/moby/blob/master/pkg/archive/archive_windows.go
+func writeHeader(tw *tar.Writer, tarHeader *tar.Header) error {
+	if runtime.GOOS == "windows" {
+		tarHeader.Mode = int64(chmodTarEntry(os.FileMode(tarHeader.Mode)))
+	}
+
+	return tw.WriteHeader(tarHeader)
+}
+
+// Code copied from https://github.com/moby/moby/blob/master/pkg/archive/archive_windows.go
+func chmodTarEntry(perm os.FileMode) os.FileMode {
+	//perm &= 0755 // this 0-ed out tar flags (like link, regular file, directory marker etc.)
+	permPart := perm & os.ModePerm
+	noPermPart := perm &^ os.ModePerm
+	// Add the x bit: make everything +x from windows
+	permPart |= 0111
+	permPart &= 0755
+
+	return noPermPart | permPart
 }

